warp-contracts-plugin-quickjs/src/eval/QuickJsEvaluator.ts

```diff
@@ -50,6 +50,42 @@ export class QuickJsEvaluator {
     randomHandle.dispose();
   }
 
+  dummyPromiseEval() {
+    const dummyPromiseEval = this.vm.newFunction('dummyPromise', () => {
+      const promise = this.vm.newPromise();
+      promise.resolve(this.vm.newString(''));
+      promise.settled.then(this.vm.runtime.executePendingJobs);
+      return promise.handle;
+    });
+    this.vm.setProp(this.vm.global, 'dummyPromise', dummyPromiseEval);
+    dummyPromiseEval.dispose();
+  }
+
+  evalExternal() {
+    const readExternalHandle = this.vm.newFunction('readExternal', (processIdHandle, actionHandle) => {
+      const promise = this.vm.newPromise();
+      this.readExternal(processIdHandle, actionHandle).then((result) => {
```

Comment:
shouldn't we also somehow handle errors here and pass some error info to the quickjs?